packages/agora-rtc-react/src/hooks/tracks.ts

```diff
@@ -127,3 +133,229 @@ export function useVolumeLevel(audioTrack?: IRemoteAudioTrack | ILocalAudioTrack
+        if (!user.audioTrack && users.some(({ uid }) => user.uid === uid)) {
+          await resolvedClient.subscribe(user, "video").catch(error => {
+            void error;
+          });
+          if (
+            user.audioTrack &&
+            !nextTracks.current.some(track => track.getUserId() === user.uid)
+          ) {
+            nextTracks.current.push(user.audioTrack);
+          }
+
+          nextTracks.current = nextTracks.current.map(track => {
+            if (
+              user.audioTrack &&
+              track.getUserId() === user.uid &&
+              track.getTrackId() !== user.audioTrack.getTrackId()
+            ) {
+              return user.audioTrack;
+            } else {
+              return track;
+            }
+          });
+          if (!isUnmounted) {
+            setTracks(nextTracks.current);
+          }
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    const unsubscribe = async (user: IAgoraRTCRemoteUser): Promise<void> => {
+      try {
+        if (users.some(({ uid }) => user.uid === uid)) {
+          if (!isUnmounted) {
+            nextTracks.current = nextTracks.current.filter(track => track.getUserId() !== user.uid);
+            console.log(nextTracks.current);
+            setTracks(nextTracks.current);
+          }
+
+          await resolvedClient.unsubscribe(user, "audio");
+        }
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    users.map(user => {
+      if (!user.audioTrack && user.hasAudio) {
+        subscribe(user);
+      }
+    });
+
+    const unsubscribeList: massUserProps[] = [];
+    for (let i = 0; i < nextTracks.current.length; i++) {
+      const track = nextTracks.current[i];
+      if (users && users.length > 0 && !users.some(user => user.uid === track.getUserId())) {
+        const user = resolvedClient.remoteUsers.find(user => user.uid === track.getUserId());
+        if (user) {
+          console.log(user.uid, "unsubscribe audio track");
+          unsubscribeList.push({
+            user,
+            mediaType: "audio",
+          });
+          nextTracks.current.splice(i, 1);
+          i--;
+        }
+      }
+    }
+    if (unsubscribeList.length > 0) {
+      await resolvedClient.massUnsubscribe(unsubscribeList);
+      if (!isUnmounted) {
+        setTracks(nextTracks.current.slice());
+      }
+    }
+
+    return joinDisposers([
+      () => {
+        isUnmounted = true;
+      },
+      listen(resolvedClient, "user-published", (pubUser, pubMediaType) => {
+        if (users.find(user => user.uid === pubUser.uid) && pubMediaType === "audio") {
+          subscribe(pubUser);
+        }
+      }),
+      listen(resolvedClient, "user-unpublished", (pubUser, pubMediaType) => {
+        if (users.find(user => user.uid === pubUser.uid) && pubMediaType === "audio") {
+          unsubscribe(pubUser);
+        }
+      }),
+    ]);
+  }, [isConnected, resolvedClient, tracks, users]);
```

Comment:
This hook does not use `tracks`, so remove it from the deps.

Reply:
fixed